### PR TITLE
New release build script (Proposal/PoC)

### LINKED ISSUE #12932: Finish Clojure release script
Finish porting the last few things/polish and PR feedback.

### PR DESCRIPTION
**Pull request by camsaul**
This PR has been moved to #13417 since I accidentally merged this one and then reverted it. Can't figure out how to reopen this PR now that it has been merged

### CODE REVIEW

**Comment by tlrobinson** [comment]
`metabase-release` already supported CE and EE builds, pre-releases, etc.

File: release/release/common.clj [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+
+(defn env-or-throw
+  "Fetch an env var value or throw an Exception if it is unset."
+  [k]
+  (or (get env/env k)
+      (throw (Exception. (format "%s is unset. Please set it and try again." (str/upper-case (str/replace (name k) #"-" "_")))))))
+
+(def ^String root-directory
+  "e.g. /Users/cam/metabase"
+  (.getParent (File. (env/env :user-dir))))
```
Comment (walterl) [inline release/release/common.clj:21]: `$USER_DIR` doesn't exist on Linux. Is this the same as `$PWD`?

Comment (camsaul) [reply release/release/common.clj:21]: Environ merges in Java system properties as well as environment variables into the `env` map, so in this case `user-dir` comes from the JVM property `user.dir`

File: release/release/check_prereqs.clj [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            (read-line)
+            (recur)))))
+    (c/announce "All required external commands are available.")))
+
+(def ^:private required-env-vars
+  (filter
+   some?
+   [:github-token
+    :dockerhub-username
+    :dockerhub-password
```
Comment (ascandella) [inline release/release/check_prereqs.clj:31]: I'm not sure if this is intentional or not, but the other scripts try to `docker login` with these credentials. I actually prefer if that's taken care of **outside** of this script, which means these variables are unused / unnecessary.

File: release/release/validate.clj [hunk 3]
```diff
@@ -0,0 +1,7 @@
+(ns release.validate)
+
+(defn validate-release []
```
Comment (ascandella) [inline release/release/validate.clj:3]: TODO

File: resources/locales.clj [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
-  :locales  #{"en" "zh" "zh-HK" "zh-TW" "uk" "fr" "nl" "tr" "sk" "it" "es" "nb" "ca" "de" "fa" "ja" "vi" "pl" "pt" "ru" }
+  :locales  #{"en" "nl" "zh" "zh-HK" "fr" "ja" "de" "es" "ca" "pl" "zh-TW" "it" "ru" "fa" "sk" "pt" "tr" }
```
Comment (ascandella) [inline resources/locales.clj:2]: Is this related to this change?

Comment (camsaul) [reply resources/locales.clj:2]: Running the build script ends up changing `locales.clj`. Commiting this was accidental. Will revert

**Comment by camsaul** [comment]
I still want to get this build script polished and ready for prime-time, but it's not there yet. So for 0.36.0 we can continue to use the bash script. For a future release I'm hoping to have this ready

**Comment by camsaul** [comment]
Oops. I did not mean to merge this into master -- not sure how that happened. Will revert and merge once finished

**Pull request merged**